stable/pomerium/values.yaml

```diff
@@ -0,0 +1,111 @@
+# For detailed explanation of each of the configuration settings see
+# https://www.pomerium.io/docs/config-reference.htmls
+
+# settings that are shared by all services
+config:
+  # secrets can be generated with `head -c32 /dev/urandom | base64`
+  sharedSecret: "LL3g/NoKkao7mTec2/wYSOZAki1VbR89q+kqbdFj0c4="
+  cookieSecret: "tAhd5A5Y8R/H/msFhkNdt8dCym2SatVbh4UrcDk3+Ms="
+  # default is to use a self-signed cert, should _NOT_ be used inproduction
+  key: "LS0tLS1CRUdJTiBFQyBQUklWQVRFIEtFWS0tLS0tCk1IY0NBUUVFSUtzZ3pUNDdBQXlzTkJNa1BJYnVJV1BodFEwN2RzWGx4WjB5Z3pjV3VXaVVvQW9HQ0NxR1NNNDkKQXdFSG9VUURRZ0FFLzMyY2c0YWpuWHpyT2pORjlXczROK2NyUTdwTGVWOGt6Y3R6cGN1S2F1KzAzVmZXRlRaTQpaTmh2cFQrUzZlS3QzSW1ycExsRTlKRFN6dDZwcUI0UVZRPT0KLS0tLS1FTkQgRUMgUFJJVkFURSBLRVktLS0tLQo="
+  cert: "LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUJFekNCdWdJSkFLK3VPcmsyVzFRTk1Bb0dDQ3FHU000OUJBTUNNQkV4RHpBTkJnTlZCQU1NQm5WdWRYTmwKWkRBZ0Z3MHhPVEF6TWpBeU1UTTNNemRhR0E4ME56VTNNREl4TkRJeE16Y3pOMW93RVRFUE1BMEdBMVVFQXd3RwpkVzUxYzJWa01Ga3dFd1lIS29aSXpqMENBUVlJS29aSXpqMERBUWNEUWdBRS8zMmNnNGFqblh6ck9qTkY5V3M0Ck4rY3JRN3BMZVY4a3pjdHpwY3VLYXUrMDNWZldGVFpNWk5odnBUK1M2ZUt0M0ltcnBMbEU5SkRTenQ2cHFCNFEKVlRBS0JnZ3Foa2pPUFFRREFnTklBREJGQWlFQTQwSjYycmJjcThYK0EzdTdQemRBNjRhaTJFbWhMQzZkRERYaQpwNmZpaHhFQ0lDRGFWcVkrUHFmZG5Ja1ZQVTFWdU5zNDl5RkJyd2IxYTlqcUdWQzlLaG1GCi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K"
+  policyFile: |-
+    - from: httpbin.corp.pomerium.io
+      to: http://httpbin
+      allowed_domains:
+      - pomerium.io
+    - from: external-httpbin.corp.pomerium.io
+      to: httpbin.org
+      allowed_domains:
+      - gmail.com
+    - from: weirdlyssl.corp.pomerium.io
+      to: http://neverssl.com
+      allowed_users:
+      - bdd@pomerium.io
+      allowed_groups:
+      - admins
+      - developers
+    - from: hello.corp.pomerium.io
+      to: http://hello:8080
+      allowed_groups:
+      - admins
+
+authenticate:
+  name: authenticate
+  redirectUrl: https://auth.corp.pomerium.io/oauth2/callback
+  proxyRootDomains: pomerium.io
+  # see https://www.pomerium.io/docs/identity-providers.html
+  idp:
+    provider: google
+    # url: https://example.oktapreview.com/oauth2/default
+    clientID: ""
+    clientSecret: ""
+
+authorize:
+  name: authorize
+
+proxy:
+  name: proxy
+  authenticateServiceUrl: https://auth.corp.pomerium.io
+  authorizeServiceUrl: https://access.corp.pomerium.io
+
+# For any other settings that are optional. for a complete listing see:
+# https://www.pomerium.io/docs/config-reference.html
+extraEnv: {}
+extraArgs: {}
+extraVolumes: {}
+
+image:
+  repository: "pomerium/pomerium"
+  tag: "latest"
+  pullPolicy: "IfNotPresent"
+
+service:
+  # Service type can be set to ClusterIP, NodePort or LoadBalancer.
+  type: ClusterIP
+  externalPort: 443
```

Comment:
@desimone is it mandatory to use https or is it possible to expose a non-https proxy so that ssl termination can be handled by nginx-ingress for example? If yes, the chart should allow for it I believe

Reply:
TLS is a requirement for all communication. Both between services and behind the ingress. I will better document the process in https://github.com/pomerium/pomerium/pull/65

Reply:
ah ok, that's what I was missing! And is this going to stay a hard requirement? We could imagine doing http when staying behind the ingress hypothetically, especially to avoid doing double work with the ingress doing also http.

Reply:
Totally fair question, and probably is something I should be doing a better job of explaining in the docs. 

So to give a little bit more more context here...  One of the reasons I started working on pomerium was that I felt that the assumptions baked into a perimeter security posture -- that "things behind the firewall were secure" --  was flawed. Reading the beyondcorp papers, and learning more about Google's aurora, and NSA breach reinforced the importance of adopting a zero-trust stance, even within your own (or cloud) datacenter, when it comes to service communication. Because of that, I feel the ideal is to mutually authenticate and encrypt between all services even behind the ingress. 

Of course, that decision has trade-offs. As you mentioned, we are doing double duty on our TLS-terminating load balancer. Also, we have to setup maintain PKI to handle the encryption. 

So on one hand, I totally understand wanting to do TLS termination, and gain the performance benefit and management ease of not having to handle TLS connections, but on the other, if pomerium is going to be a "zero-trust" beyondcorp-inspired solution, and not just an authenticating reverse proxy, it feels like mTLS is a cornerstone that. 

tl;dr; - I think the security win is worth the performance hit, and we can improve the UX of the mTLS so it isn't so much of a burden. But I am very open to feedback.

Reply:
Thank for taking the time to explain, this makes total sense. And anyway, this is not blocking I believe :)

Reply:
I added some logic to bootstrap the certificate/ca process.